zds/tutorialv2/views/contents.py

```diff
@@ -1287,9 +1287,8 @@ def form_valid(self, form):
 
         self.success_url = extract.get_absolute_url()
         if self.request.is_ajax():
-            return HttpResponse(json.dumps({'result': 'ok', 'last_hash': extract.compute_hash(),
-                                            'new_url': extract.get_edit_url()}),
-                                content_type='application/json')
+            return JsonResponse(json.dumps({'result': 'ok', 'last_hash': extract.compute_hash(),
+                                            'new_url': extract.get_edit_url()}))
```

Comment:
`JsonResponse` prend directement un dictionnaire (ou autres) en paramètres, pas une chaîne JSON, et se charge de la sérialisation :) .

```suggestion
            return JsonResponse({'result': 'ok', 'last_hash': extract.compute_hash(),
                                 'new_url': extract.get_edit_url()})
```